### DISCUSSION TITLE
How to create ExtensionContext for writing tests on an extension

### DISCUSSION THREAD

**Discussion by khmarbaise** [junit-team/junit-framework#3020] [Q&A] [answered]
I'm currently in the situation having an extension for JUnit JUpiter which works so far fine..but I'm trying to find a way to create an ExtensionContext as given during the execution of junit jupiter (which is injected into the extension)  based on a particular class. The intention is to write some tests which need the ExtensionContext. For example in my extension I have defined:

```java
@Override
public void beforeTestExecution(ExtensionContext context) {
...
}
```
So the question is: Is there a way to create the `ExtensionContext` based on a give class for example, having the following test class:

```java
@MavenProfile("profile-on-class")
class ProfileAnnotationOnClass {

  @MavenTest
  void name() {

  }
}
```

I don't want to really execute the tests I only want to have a correct `ExtensionContext` which is used for analysing several things like the annotation... 
So this would make it possible to write better tests for particular parts of my extension?

**Comment by sbrannen** [comment]
> Is there a way to create the `ExtensionContext` based on a given class?

No, that's not possible.

You can either mock the `ExtensionContext`, or you can use the `EngineTestKit` to execute a single test class or test method.

We recommend using the `EngineTestKit` to test JUnit Jupiter extensions, and we do that often in our own test suite.

**Comment by khmarbaise** [comment]
Hi @sbrannen, many thanks for your hints. 

Do you have small hint where I can find such examples for using `EngineTestKit` for extensions?

**Reply by sbrannen** [reply]
> Hi @sbrannen, many thanks for your hints.

You're welcome.

> Do you have small hint where I can find such examples for using `EngineTestKit` for extensions?

There are many examples in JUnit 5's own code base.

However, I'm currently working on Spring and found this one rather quickly:

https://github.com/spring-projects/spring-framework/blob/main/spring-test/src/test/java/org/springframework/test/context/junit/jupiter/DisabledIfAndDirtiesContextTests.java

**Comment by khmarbaise** [comment, marked as answer]
Many thanks for your hints.
